rn/Teacher/src/modules/submissions/list/get-submissions-props.js

```diff
@@ -142,45 +128,85 @@ export function pendingProp (assignmentContent?: AssignmentContentState, courseC
   return assignmentContent.submissions.pending > 0 || courseContent.enrollments.pending > 0
 }
 
+// If a submission does not have a grade or an actual submission it will not have a group attached to the submission.
+// So if it's there we get the group from entities, but if not then we will look up the group in entities based on
+// the category id and if the user is in the group.
+export function getGroup (groups: GroupsState, submission: SubmissionWithHistory, categoryID: ?string): ?Group {
+  if (submission.group && submission.group.id != null) {
+    let group = groups[submission.group.id]
+    if (!group) return
+    return group.group
+  }
+
+  if (!categoryID) return
+
+  // $FlowFixMe
+  let groupValues: Array<{ group: Group }> = Object.values(groups)
+  let group = groupValues
+    .filter(({ group }) => group && group.group_category_id === categoryID)
```

Comment:
This is where we are now checking to make sure that the group is still a known group which will save us from regressing on [this](https://github.com/instructure/ios/pull/1448), right?

Reply:
No that is in the map-state-to-props where we are checking to make sure the group exists to say it's a groupGraded assignment.